src/reporting/clients/IMappingsClient.ts

```diff
@@ -0,0 +1,537 @@
+
+  /**
+   * Gets a Mapping for an iModel.
+   * @param {string} iModelId The iModel Id.
+   * @param {string} mappingId The Mapping Id.
+   * @param {string} accessToken OAuth access token with scope `insights:read`
+   * @memberof ReportingClient
+   * @link https://developer.bentley.com/apis/insights/operations/get-mapping/
+   */
+  getMapping(accessToken: AccessToken,
+    iModelId: string,
+    mappingId: string
+  ): Promise<Mapping>;
+
+  /**
+   * Creates a Mapping for an iModel.
+   * @param {string} iModelId Id of the iModel for which to create a new Mapping.
+   * @param {string} accessToken OAuth access token with scope `insights:modify`
+   * @param {MappingCreate} mapping Request body.
+   * @memberof ReportingClient
+   * @link https://developer.bentley.com/apis/insights/operations/create-mapping/
+   */
+  createMapping(accessToken: AccessToken,
+    iModelId: string,
+    mapping: MappingCreate
+  ): Promise<Mapping>;
+
+  /**
+   * Updates a Mapping for an iModel.
+   * @param {string} iModelId The iModel Id.
+   * @param {string} mappingId Id of the Mapping to be updated.
+   * @param {string} accessToken OAuth access token with scope `insights:modify`
+   * @param {MappingUpdate} mapping Request body.
+   * @memberof ReportingClient
+   * @link https://developer.bentley.com/apis/insights/operations/update-mapping/
+   */
+  updateMapping(accessToken: AccessToken,
+    iModelId: string,
+    mappingId: string,
+    mapping: MappingUpdate
+  ): Promise<Mapping>;
+
+  /**
+   * Deletes a Mapping for an iModel.
+   * @param {string} iModelId The iModel Id.
+   * @param {string} mappingId Id of the Mapping to be deleted.
+   * @param {string} accessToken OAuth access token with scope `insights:modify`
+   * @memberof ReportingClient
+   * @link https://developer.bentley.com/apis/insights/operations/delete-mapping/
+   */
+  deleteMapping(accessToken: AccessToken,
+    iModelId: string,
+    mappingId: string
+  ): Promise<Response>;
+
+  /**
+   * Copies a Mapping and all its Groups, GroupProperties, CalculatedProperties, and CustomCalculations to a target iModel.
+   * @param {string} iModelId Id of the source Mapping&#x27;s iModel.
+   * @param {string} mappingId Id of the source Mapping.
+   * @param {string} accessToken OAuth access token with scope `insights:modify`
+   * @param {MappingCopy} mappingCopy Request body.
+   * @memberof ReportingClient
+   * @link https://developer.bentley.com/apis/insights/operations/copy-mapping/
+   */
+  copyMapping(accessToken: AccessToken,
+    iModelId: string,
+    mappingId: string,
+    mappingCopy: MappingCopy
+  ): Promise<Mapping>;
+
+  /**
+   * Gets all Groups for a Mapping. This method returns the full list of groups.
+   * @param {string} iModelId The iModel Id.
+   * @param {string} mappingId The Mapping Id.
+   * @param {string} accessToken OAuth access token with scope `insights:read`
+   * @param {number} top The number of entities to load.
```

Comment:
`The number of entities to load per page.`

Reply:
comments changed